src/nccl_ofi_rdma.cpp

```diff
@@ -2752,38 +2771,34 @@ int nccl_net_ofi_rdma_domain_t::reg_internal_mr(void *data,
 	return this->reg_mr(&ckey, type, mhandle);
 }
 
-static int reg_mr_send_comm(nccl_net_ofi_send_comm_t *send_comm,
-			    nccl_ofi_mr_ckey_ref ckey,
-			    int type, void **mhandle)
+int nccl_net_ofi_rdma_send_comm_t::regMr(nccl_ofi_mr_ckey_ref ckey, int mr_type, void **mhandle)
 {
-	nccl_net_ofi_rdma_ep_t *ep = (nccl_net_ofi_rdma_ep_t *)send_comm->base.ep;
-        nccl_net_ofi_rdma_domain_t *domain = ep->rdma_endpoint_get_domain();
-	assert(domain != NULL);
+	auto *ep_ptr = this->rdma_send_comm_get_ep();
+        nccl_net_ofi_rdma_domain_t *domain = ep_ptr->rdma_endpoint_get_domain();
+	assert(domain != nullptr);
 
 	pthread_wrapper domain_lock(&domain->domain_lock);
 
-	CHECK_DOMAIN_ACTIVE(domain, "reg_mr_send_comm");
+	CHECK_DOMAIN_ACTIVE(domain, "send regMr");
```

Comment:
Note: to differentiate between regMr function calls by send comms and recv comms, I passed in "<send/recv> regMr" as an arg to CHECK_DOMAIN_ACTIVE (even though it doesn't exactly match the function name like before) since the arg is only used for a print.